Jonesy22/Perro
# Backend server up

The backend server is located in the `express-backend` folder. The backend will start on port 5000 and serve the API for the frontend, which is on port 3000 after running `npm start`. Something I forgot was that the backend and frontend app will be two different terminal sessions and be simultaneously running as the frontend makes requests to the backend to actually save and load data (where the database is).

This is only the beginning of the backend and more endpoints will need to be added. For example, we need an endpoint to GET a single user or GET a project. As of now, there is only a single endpoint as a GET to retrieve all users in a `dummyDatabase.js`.

**Test Plan**
- Checkout this `PER-16-backend-express-server` branch.
- Open two terminal sessions and run `npm start` in one (frontend) and `nodemon express-backend/index.js` in the other (backend). Note: nodemon is a module that allows changes made in the backend to automatically restart the server.
- Ensure the Perro frontend loads in the browser without error.
- In the browser, hit the endpoint `/list` with the URL: `http://localhost:5000/users/list`. You should get a list of users that I hardcoded in `dummyDatabase.js`. Later, this will be served through our database. I followed the schema Daniel made in LucidChart (https://lucid.app/lucidchart/cc87fd8c-cb70-4b67-b4c7-1d0dfb56c31a/edit?useCachedRole=false&shared=true&page=0_0#).
